File: data/multilang_comparison/program_julia/program_julia/src/func36.jl
include("program_julia_head.jl")
function func36(vars::JArray_param, PATH0::UInt64, loopsFactor::Int)::JArray
   pCounter = length(vars.data)
   if (PATH0 & 1) != 0
      params0 = JArray_param{Int}()
      params0.size = 0
      array70 = func37(params0, get_path(), loopsFactor);
      array70.refC -= 1
      if array70.refC == 0
          empty!(array70.data)
          array70 = nothing
      end
   
   else 
      params0 = JArray_param{Int}()
      params0.size = 0
      array76 = func38(params0, get_path(), loopsFactor);
      array76.refC -= 1
      if array76.refC == 0
          empty!(array76.data)
          array76 = nothing
      end
   end
   params0 = JArray_param{Int}()
   params0.size = 0
   array81 = func39(params0, get_path(), loopsFactor);
   return array81;
end

File: data/multilang_comparison/program_go/program_go/src/func36.go
package main
import "math/rand"
var _ = rand.Intn
func Func36(vars *ArrayParam, PATH0 uint64 , loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   if (PATH0 & 1) != 0 {
      var params0 ArrayParam
      params0.Size = 0
      params0.Data = make([]*Array, params0.Size)
      var array70 *Array = Func37(&params0, getPath(), loopsFactor);
      params0.Data = nil
      params0.Size = 0
      array70.RefC--
      if array70.RefC == 0 {
          array70.Data = nil
          array70 = nil
      }
   
   } else {
      var params0 ArrayParam
      params0.Size = 0
      params0.Data = make([]*Array, params0.Size)
      var array76 *Array = Func38(&params0, getPath(), loopsFactor);
      params0.Data = nil
      params0.Size = 0
      array76.RefC--
      if array76.RefC == 0 {
          array76.Data = nil
          array76 = nil
      }
   }
   var params0 ArrayParam
   params0.Size = 0
   params0.Data = make([]*Array, params0.Size)
   var array81 *Array = Func39(&params0, getPath(), loopsFactor);
   params0.Data = nil
   params0.Size = 0
   return array81;
}

